### PR TITLE
fix DMA unmap dirty page bitmap section

### PR DESCRIPTION
**Pull request by tmakatos**
Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

File: docs/vfio-user.rst [hunk 1]
```diff
@@ -702,16 +704,23 @@ following format:
...
-The VFIO bitmap structure is defined in ``<linux/vfio.h>``
-(``struct vfio_bitmap``).
 
-Each ``struct vfio_bitmap`` entry is followed by the region's bitmap. Each bit
-in the bitmap represents one page of size ``struct vfio_bitmap.pgsize``.
+.. Note::
+   The size of the bitmap can be larger than necessary. The server must still
+   respond with the same bitmap size, zeroing the unused part.
+
+Response
```
Comment (jlevon) [inline docs/vfio-user.rst:712]: nit, we use "Reply" elsewhere (and I'm doing so in my rewrite)

File: docs/vfio-user.rst [hunk 2]
```diff
@@ -702,16 +704,23 @@ following format:
...
+^^^^^^^^
+
+The server responds with the standard vfio-user header. For each DMA region
+that has the ``VFIO_DMA_UNMAP_FLAG_GET_DIRTY_BITMAP`` bit is set in Flags in
```
Comment (jlevon) [inline docs/vfio-user.rst:716]: nit, /is set/set/

File: docs/vfio-user.rst (hunk 1 above)
Comment (tmakatos) [reply docs/vfio-user.rst:712]: ok

File: docs/vfio-user.rst (hunk 2 above)
Comment (tmakatos) [reply docs/vfio-user.rst:716]: ok

File: docs/vfio-user.rst [hunk 3]
```diff
@@ -661,33 +661,35 @@ Table entry format
...
     must be populated before unmapping the DMA region. The client must provide
-    a ``struct vfio_bitmap`` in the VFIO bitmaps field for each region, with
-    the ``vfio_bitmap.pgsize`` and ``vfio_bitmap.size`` fields initialized.
+    a ``struct vfio_user_bitmap`` (explained below) in the VFIO bitmap field
+    for the DMA region, with the ``pgsize`` and ``size`` fields initialized.
+
+* *VFIO Bitmap* contains a ``struct vfio_user_bitmap`` for the DMA region if
+  ``VFIO_DMA_UNMAP_FLAG_GET_DIRTY_BITMAP`` is set in Flags.
+
+The position of the next DMA region in the table is therefore dependant on
```
Comment (jlevon) [inline docs/vfio-user.rst:681]: nit, "dependent"

**Review by jlevon — CHANGES_REQUESTED** [review]
just one nit, thanks.

File: docs/vfio-user.rst (hunk 3 above)
Comment (tmakatos) [reply docs/vfio-user.rst]: ok

**Review by jlevon — APPROVED** [review]

**Pull request merged**